.github/workflows/release.yml

```diff
@@ -38,3 +38,23 @@ jobs:
           name: ${{ steps.semantic_release_info.outputs.git_tag }}
           body: ${{ steps.semantic_release_info.outputs.notes }}
           target_commitish: ${{ steps.commit_and_push.outputs.latest_commit_hash }}
+
+      - name: Go Mod Tidy
+        if: ${{ steps.semantic_release_info.outputs.git_tag }}
+        run: go mod tidy
+
+      - name: Go Build
+        if: ${{ steps.semantic_release_info.outputs.git_tag }}
+        run: CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w -X github.com/checkmarx/2ms/cmd.Version=${{ steps.semantic_release_info.outputs.git_tag }}" -a -installsuffix cgo -o bin/2ms main.go
+
+      - name: Upload binaries to release
+        if: ${{ steps.semantic_release_info.outputs.git_tag }}
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: bin/2ms
+          asset_name: 2ms
+          tag: ${{ steps.semantic_release_info.outputs.git_tag }}
+          overwrite: true
+          body: "Version Generated"
```

Comment:
you can remove this part as `softprops/action-gh-release` step has option to upload files

Reply:
the thing is that `softprops/action-gh-release` is happening before the build, and we need the `steps.semantic_release_info.outputs.git_tag` that comes from it (since is the latest released version) and then during the build we are injecting the `steps.semantic_release_info.outputs.version` in the new binary file so we can access that info in the code

Reply:
reposition `- name: Go Build` step before the `- name: Creating Release`
the step creating the `steps.semantic_release_info.outputs.git_tag` is `- name: Gets release info` (at the top)